src/ctapipe/tools/tests/test_train.py

```diff
@@ -104,12 +104,12 @@ def test_signal_fraction(tmp_path, gamma_train_clf, proton_train_clf):
             ],
         )
 
-        with open(log_file, "r") as f:
+        with open(log_file) as f:
             log = f.readlines()
 
         for line in log[::-1]:
             if "Train on" in line:
-                n_signal, n_background = [int(line.split(" ")[i]) for i in (7, 10)]
+                n_signal, n_background = (int(line.split(" ")[i]) for i in (7, 10))
```

Comment:
changing from list to tuple, not sure that's desired

Reply:
That's not changing from a list to a tuple, it's changing from a list comprehension to a generator expression.

I didn't do this manually, it was done by `ruff --select=UP --fix`, and it avoids the intermediate list that is directly unpacked into the two variables